### PR TITLE
fix: mage installation with go1.18

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

go1.18 changed where the packages are installed when using `go install`
Run the same command in the CI pipeline

## Why is it important?

The UR for 7.17 is broken for the last 5 days.

```
Installing mage v1.13.0.
make[3]: Leaving directory '/var/lib/jenkins/workspace/elastic+unified-release+master+snapshot-beats-7.17/cd/release/release-manager/project-configs/7.17/build/projects/gopath/src/github.com/elastic/beats/libbeat'
bash: mage: command not found
make[3]: [../dev-tools/make/mage-install.mk:11: mage] Error 127 (ignored)
bash: mage: command not found
make[3]: *** [scripts/Makefile:33
```

## Test

```bash
echo  1.17.10 > .go-version
./dev-tools/run_with_go_ver make test-mage 
...

## Variables

GOOS             = darwin
GOARCH           = amd64
GOARM            = 
Platform         = darwin/amd64
BinaryExt        = 
XPackDir         = ../x-pack
```

```bash
echo  1.18.5 > .go-version
./dev-tools/run_with_go_ver make test-mage 
...

## Variables

GOOS             = darwin
GOARCH           = amd64
GOARM            = 
Platform         = darwin/amd64
BinaryExt        = 
XPackDir         = ../x-pack
BeatName         = beats
BeatServiceName  = beats
## Functions

beat_doc_branch              = 7.17
beat_version                 = 7.17.7
commit                       = 0390d5708dec5ef6c7c721af0c29e6a4bc28261d
date                         = 2022-09-14T17:37:17Z
elastic_beats_dir            = /Users/vmartinez/work/src/github.com/elastic/beats
go_version                   = 1.18.5
```

Additionally, I tested on docker, to be the more platform agnostic as possible, see https://github.com/elastic/beats/pull/33083#issuecomment-1247083355

## Issues

Caused by https://github.com/elastic/beats/pull/32693

### CODE REVIEW

**Review by cmacknz — APPROVED** [review]
Thanks! I noticed this errors in the logs but never got around to figuring out why they were there.

**Comment by v1v** [comment]
I'm still fixing it ...

```bash
$ docker run --rm -ti -v `pwd`:/src/github.com/elastic/beats -w /src/github.com/elastic/beats ubuntu
$ apt-get update ; apt-get install -y curl make build-essential  ; HOME=/src ./dev-tools/run_with_go_ver make test-mage
+ /src/bin/gvm 1.17.10
+ cut -d ' ' -f 2
+ tr -d '\"'


++ /src/bin/gvm 1.17.10
+ eval 'export GOROOT="/src/.gvm/versions/go1.17.10.linux.amd64"
export PATH="/src/.gvm/versions/go1.17.10.linux.amd64/bin:$PATH"'
++ export GOROOT=/src/.gvm/versions/go1.17.10.linux.amd64
++ GOROOT=/src/.gvm/versions/go1.17.10.linux.amd64
++ export PATH=/src/.gvm/versions/go1.17.10.linux.amd64/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
++ PATH=/src/.gvm/versions/go1.17.10.linux.amd64/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
...

## Functions

beat_doc_branch              = {{ beat_doc_branch }}
beat_version                 = {{ beat_version }}
commit                       = {{ commit }}
date                         = {{ date }}
elastic_beats_dir            = {{ elastic_beats_dir }}
go_version                   = {{ go_version }}
repo.RootImportPath          = {{ repo.RootImportPath }}
repo.CanonicalRootImportPath = {{ repo.CanonicalRootImportPath }}
repo.RootDir                 = {{ repo.RootDir }}
```

File: Jenkinsfile
Comment (v1v) [inline Jenkinsfile:93]: This works 

image

**Comment by v1v** [comment]
@cmacknz , I think this will be ready to be merged, somehow my initial thoughts were about the GOPATH, but the issue was related to `go install`. I'll be offline in a bit, so this PR could be merged as soon as it gets green, you don't need to wait for me, in case the release team wanna build `7.17` sooner than later, otherwise I'll merge it first thing tomorrow morning

**Review by cmacknz — APPROVED** [review]

**Comment by v1v** [comment]
@mergifyio backport 8.5

**Comment by v1v** [comment]
@Mergifyio backport 8.5

**Pull request merged**
